Azure/sonic-buildimage
# [TPID CONFIG]TPID attribute Yang model and default TPID for Minigraph to configDB Changes

#### Why I did it
This PR is part of the TPID CONFIG SONiC feature set to support port/LAG TPID configuration.  
TPID HLD PR: (https://github.com/Azure/SONiC/pull/681)

There are total of 3 PRs across different submodules to achieve this feature support.

- SONiC-Utilities (https://github.com/Azure/sonic-utilities/pull/1618) - To handle TPID CLI related changes.
- SONiC-Swss (https://github.com/Azure/sonic-swss/pull/1747) - To handle Orchagent changes triggered from TPID configuration changes to interact with SYNCd SAI.
- and this PR which handles the new port/LAG TPID attribute on Yang model and Minigraph to configDB changes.

Please refer to the TPID HLD listed above for more details on the TPID configuration feature support.
Here is a quick summary for this PR changes:
By default TPID for port/LAG is 0x8100.  With HW SKU that has capability to handle TPID configuration user is allowed to configure the port/LAG to non default TPID values such as (0x9100, 0x9200, 0x88A8). 
This PR is to define this new TPID attribute for Port and LAG in the Yang model as well as to ensure all the ports/LAGs in the config DB generated from Minigraph should have a default TPID value of 0x8100 .

#### How to verify it
Test code were added for the Yang model to test the new TPID attribute in PORT and Portchannel.
Test code were also updated in test config gen, minigraph as well as multi-asic config gen to ensure the expected default TPID is in place.
Complete feature testing of TPID functionality were also performed on DUTs that has SAI support for TPID attribute to ensure the feature is working properly.

#### Which release branch to backport (provide reason below if selected)


#### Description for the changelog
TPID port/LAG attribute Yang model and default TPID for Minigraph to configDB Changes


#### A picture of a cute animal (not mandatory but encouraged)
